### PR TITLE
doc: tm_timecmp documentation.

### PR DESCRIPTION
**Pull request by i4ki**

### CODE REVIEW

**Review by soerenmartius — COMMENTED** [review]
LGTM, @i4ki this needs to be added to the sidebar also

File: cli/reference/functions/tm_timecmp.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+title: tm_timecmp | Functions | Configuration Language
+description: The tm_timecmp compares two timestamps and returns a number that represents the ordering of the instants those timestamps represent.
+---
+
+# `tm_timecmp` Function
+
+The `tm_timecmp` compares two timestamps and returns a number that represents the ordering of the instants those timestamps represent.
```
Comment (AnnuCode) [inline cli/reference/functions/tm_timecmp.md:8]:
```suggestion
The `tm_timecmp` function compares two timestamps and returns a value that indicates whether one timestamp occurs before, at the same time, or after the other.
```

File: cli/reference/functions/tm_timecmp.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+| Condition                                          | Return Value |
+| -------------------------------------------------- | ------------ |
+| `timestamp_a` is before `timestamp_b`              | -1           |
+| `timestamp_a` is the same instant as `timestamp_b` | 0            |
+| `timestamp_a` is after `timestamp_b`               | 1            |
+
+When comparing the timestamps, `tm_timecmp` takes into account the UTC offsets given in each timestamp. For example, `06:00:00+0200` and `04:00:00Z` are the same instant after taking into account the `+0200` offset on the first timestamp.
+
+In the Terramate language, timestamps are conventionally represented as strings using *RFC 3339* "Date and Time format" syntax. The `tm_timecmp` requires the its two arguments to both be strings conforming to this syntax.
```
Comment (AnnuCode) [inline cli/reference/functions/tm_timecmp.md:22]:
```suggestion
In Terramate, timestamps are represented as strings following the *RFC 3339* date and time format. The `tm_timecmp` function requires both of its arguments to be strings that adhere to this format.
```

File: cli/reference/functions/tm_timecmp.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+1
+
+tm_timecmp("2017-11-22T01:00:00Z", "2017-11-22T00:00:00-01:00")
+0
+```
+
+## Related Functions
+
+* [`tm_formatdate`](./tm_formatdate.md) can convert the resulting timestamp to
+  other date and time formats.
```
Comment (AnnuCode) [inline cli/reference/functions/tm_timecmp.md:43]:
```suggestion
* [`tm_formatdate`](./tm_formatdate.md) converts the resulting timestamp to
  other date and time formats.
```

File: cli/reference/functions/tm_timestamp.md [hunk 4]
```diff
@@ -37,3 +37,6 @@ tm_timestamp()
 
 * [`tm_formatdate`](./tm_formatdate.md) can convert the resulting timestamp to
   other date and time formats.
```
Comment (AnnuCode) [inline cli/reference/functions/tm_timestamp.md:39]:
```suggestion
* [`tm_formatdate`](./tm_formatdate.md) converts the resulting timestamp to
  other date and time formats.
```

**Review by AnnuCode — APPROVED** [review]

**Pull request merged**